src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -235,13 +241,77 @@ public static Homework parseIndividualPieceOfHomework(String individualHomework)
     /**
      * Parses {@code Collection<String> homework} into a {@code Set<Homework>}.
      */
-    public static Set<Homework> parseHomeworkList(Collection<String> homework)
-            throws ParseException {
+    public static Set<Homework> parseHomeworkList(Collection<String> homework) throws ParseException {
         requireNonNull(homework);
         final Set<Homework> homeworkSet = new HashSet<>();
         for (String description : homework) {
             homeworkSet.add(parseIndividualPieceOfHomework(description));
         }
         return homeworkSet;
     }
+
+    /**
+     * Parses {@code String keywords} into a {@code List<String>}.
+     * Leading and trailing whitespaces will be stripped and
+     * keywords string is split by whitespace into a list of words.
+     *
+     * @throws ParseException if the given {@code address} is invalid.
+     */
+    public static List<String> parseKeywords(String keywords) throws ParseException {
+        requireNonNull(keywords);
+        String strippedKeywords = keywords.strip();
+        if (strippedKeywords.isEmpty()) {
+            throw new ParseException(FindCommand.MESSAGE_KEYWORD_CONSTRAINTS);
+        }
+        return Arrays.asList(strippedKeywords.split("\\s+"));
+    }
+
+    /**
+     * Parses {@code String keywords} into a {@code String}.
+     * Leading and trailing whitespaces of keywords will be stripped.
+     * A tag keyword can only be one word.
+     *
+     * @throws ParseException if the given {@code address} is invalid.
+     */
+    public static String parseTagKeyword(String keyword) throws ParseException {
+        requireNonNull(keyword);
+        String strippedKeyword = keyword.strip();
+        if (strippedKeyword.isEmpty()) {
+            throw new ParseException(FindCommand.MESSAGE_KEYWORD_CONSTRAINTS);
+        }
+        if (!Tag.isValidTagName(strippedKeyword)) {
+            throw new ParseException(FindCommand.MESSAGE_TAG_KEYWORD_CONSTRAINTS);
+        }
+        return strippedKeyword;
+    }
+
+    /**
+     * Parses {@code Collection<String> keywords} into a {@code List<String>}.
+     */
+    public static List<String> parseTagKeywords(Collection<String> keywords) throws ParseException {
+        requireNonNull(keywords);
+        final List<String> tagKeywords = new ArrayList<>();
+        for (String keyword : keywords) {
+            tagKeywords.add(parseTagKeyword(keyword));
+        }
+        return tagKeywords;
+    }
+
+    /**
+     * Parses a {@code String condition} into a {@code FindCondition}.
```

Comment:
Is "condition" meant to be inside the {@code String}?

Reply:
is it not supposed to? I just follow homework